_posts/2018/2018-08-05-delegate-to-simplify-your-code.md

```diff
@@ -0,0 +1,90 @@
+---
+title: Delegate to simplify your code
+description: 'There’s both a Ruby and a Rails version'
+layout: article
+category: ruby
+image:
+  base: '2018/delegate-to-simplify-your-code'
+  alt: 'Graffiti arrows'
+  source: 'https://unsplash.com/photos/Ze3l02aRwmE'
+  credit: 'Annie Spratt'
+---
+
+One way of thinking about object-oriented programming is as passing messages between objects.
+
+In many cases you may want to surface a public method method on a associated object as if it was a method on the original object.
+
+The two main ways to do this are:
+
+ * the `Forwardable` functionality in the the standard Ruby library, [documentation here](http://ruby-doc.org/stdlib-2.5.1/libdoc/forwardable/rdoc/Forwardable.html#method-i-def_delegator).
+ * an Active Support core extention, `delegate`, if you’re using Rails, [documentation here](https://github.com/rails/rails/blob/master/activesupport/lib/active_support/core_ext/module/delegation.rb).
```

Comment:
This sentence might read better as:

* `delegate`, an Active Support core _extension_ (emphasis due to spelling error)..then add in parentheses after: Only available if you're using Rails — or something like that.